website/docs/automation/gitops/argocd/app-of-apps/index.md

```diff
@@ -0,0 +1,113 @@
+---
+title: "Argo CD App of Apps"
+chapter: true
+sidebar_position: 100
+---
+
+[Argo CD](https://argoproj.github.io/cd/) can deploy a set of applications to different environments (DEV, TEST, PROD ...) using common set of Kubernetes manifest and customizations specific to an environment.
+
+We can leverage [Argo CD App of Apps pattern](https://argo-cd.readthedocs.io/en/stable/operator-manual/cluster-bootstrapping/) to implement this use case. This pattern allows us to specify one Argo CD Application that consists of other applications.
+
+![argo-cd-app-of-apps](assets/argocd-app-of-apps.png)
+
+We reference [EKS Workshop Git repository](https://github.com/aws-samples/eks-workshop-v2/tree/main/environment/workspace/manifests) as a Git repository with Common Manifest. This repository will contain an initial resource state for each application.
+
+```
+.
+|-- manifests
+| |-- assets
+| |-- carts
+| |-- catalog
+| |-- checkout
+| |-- orders
+| |-- other
+| |-- rabbitmq
+| `-- ui
+```
+
+This example shows how to use Helm to create a configuration for a particular, for example DEV, environment.
+A typical layout of a Git repository could be:
+
+```
+.
+|-- app-of-apps
+|   |-- ...
+`-- apps-kustomization
+    ...
+```
+
+We create templates for each child application:
+
+```
+.
+|-- app-of-apps
+|   |-- Chart.yaml
+|   |-- templates
+|   |   |-- assets.yaml
+|   |   |-- carts.yaml
+|   |   |-- catalog.yaml
+|   |   |-- checkout.yaml
+|   |   |-- orders.yaml
+|   |   |-- other.yaml
+|   |   |-- rabbitmq.yaml
+|   |   `-- ui.yaml
+|   `-- values.yaml
+`-- apps-kustomization
+    ...
+```
+
+`Chart.yaml` is a boiler-plate. `templates` contains one file for each child application, for example:
+
+```file
+automation/gitops/argocd/app-of-apps/templates/ui.yaml
+```
+
+`values.yaml` contains values which are specific for a particular environment and which will be applied to all application templates.
+
+```file
+automation/gitops/argocd/app-of-apps/values.yaml
+```
+
+First, copy `App of Apps` configuration which we described above to the Git repository directory:
+
+```bash
+$ cd ~/environment
```

Comment:
Lets remove the `cd` command and change the `yq` command to something like:

```
yq -i ".spec.source.repoURL = load(\"$HOME/environment/argocd_repo_url\")" ~/environment/argocd/app-of-apps/values.yaml
```

Reply:
changed